server/game/cards/agendas/TheRainsOfCastamere.js

```diff
@@ -33,11 +31,20 @@ class TheRainsOfCastamere extends AgendaCard {
             target: {
                 type: 'select',
                 activePromptTitle: 'Select a plot',
-                cardCondition: card => card.controller === this.controller && card.location === 'scheme plots',
+                cardCondition: card => card.controller === this.controller && card.hasTrait('scheme'),
                 cardType: 'plot'
             },
             handler: context => this.trigger(context)
         });
+
+        this.persistentEffect({
+            targetType: 'player',
+            targetController: 'current',
+            effect: [
+                ability.effects.cannotSelectSchemes(),
+                ability.effects.groupCardPile('plot deck')
```

Comment:
This feels like the server knowing too much about the UI presentation. Shouldn't this check be done on the client as it was previously when the client would check for the agenda and then conditionally render a separate schemes pile?

It feels a lot like the `additionalPiles` mechanism (ironically, it was used for the scheme pile) we used to have that we took out because it had too much knowledge about the UI.

Reply:
I sort of agree, but on the other hand the client is supposed to be somewhat dumb about the rules engine and just do what the server tells it.  I plan to use this mechanism, or something very like it, to enable the rookery display at the start of the game.

If anything, this is more of a hint that there's a new property of the data.  It's still up to the client to decide how to display that change.

Happy to consider alternatives that don't involve the client checking for specific cards or agendas